### PR TITLE
IOS: Further deglobalization and emulation/maintainance code splitting.

### PR DESCRIPTION
**Pull request by AdmiralCurtiss**
@leoetlino You were interested in this back in #11787. I think this is a pretty good solution all things considered.

### CODE REVIEW

File: Source/Core/Core/IOS/IOS.cpp [hunk 1]
```diff
@@ -304,7 +304,7 @@ Kernel::Kernel(IOSC::ConsoleType console_type) : m_iosc(console_type)
   ASSERT(m_fs);
 
   std::lock_guard lock(m_device_map_mutex);
```
Comment (leoetlino) [inline Source/Core/Core/IOS/IOS.cpp]: I don't think we need to lock this mutex anymore

**Review by leoetlino — APPROVED** [review]
lgtm

File: Source/Core/Core/IOS/IOS.cpp (hunk 1 above)
Comment (AdmiralCurtiss) [reply Source/Core/Core/IOS/IOS.cpp]: Mhm, I suppose it doesn't make much sense to lock this in the constructor, yeah.

Comment (AdmiralCurtiss) [reply Source/Core/Core/IOS/IOS.cpp]: In fact... is this mutex needed at all? It's only locked in the constructor, destructor, and a single function (`GetDeviceByName()`).

**Pull request merged**